src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java

```diff
@@ -933,6 +934,10 @@ public synchronized void startReplica(
               verbose ? System.out : new PrintStream(OutputStream.nullOutputStream()),
               primaryGen);
 
+      if (indexState.globalState.configuration.getSyncInitialNrtPoint()) {
+        nrtReplicaNode.syncFromCurrentPrimary(INITIAL_SYNC_PRIMARY_WAIT_MS);
+      }
```

Comment:
In `ShardState.isStarted()` which is used by `IndexState.isStarted()` we consider the shard to be ready if `nrtReplicaNode != null`: https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java#L209
What do you think about assigning the `NRTReplicaNode` to a local variable, syncing the nrt point and then assigning it to the global variable so that we consider a shard ready after the sync is complete? Not sure if syncing without that global assignment causes any other issues though

Reply:
I added an explicit `started` flag, so it is no longer a surprise.